src/validation.cpp

```diff
@@ -1305,6 +1305,14 @@ bool CChainState::IsInitialBlockDownload() const
         return false;
     if (fImporting || fReindex)
         return true;
+    uint64_t ibdtimeout = gArgs.GetArg("-ibdtimeout", DEFAULT_IBD_TIMEOUT);
+    int64_t uptime = GetTime() - GetStartupTime();
+    assert(uptime >= 0);
```

Comment:
This assumes that `GetTime()` is monotonic, which I don't think is true. `GetTime()` is mockable so may move backwards in tests.

It looks like `GetTime()` is also deprecated.

Reply:
Nodes in IBD will not respond to GETHEADERS. Nodes in IBD do not request headers from inbound peers only outbound.

If all or most of the listening nodes are in IBD (emergency hot fix or something) then nobody will be able to get headers.

To get out of this the listening nodes need to request headers from inbound connections, either by exiting IBD or special casing that limit while in IBD.

Reply:
bc1que90l5dy4yzn79x8xmem8t7dgn0c2zcqvxnr6x